### PR TITLE
Panic notifications

### PR DESCRIPTION
**Pull request by alygin**
Implementation of notifications on panics as described in issue #45.

### CODE REVIEW

File: lib/cargo.js [hunk 1]
```diff
@@ -163,33 +170,114 @@ export function provideBuilder() {
...
+            });
+          }
+        }
+      }
+
+      // Tries to parse a stack trace. Returns the quantity of actually parsed lines.
+      function tryParseStackTrace(lines, i, panic) {
+        let parsedQty = 0;
+        let line = lines[i];
+        if (line.substring(0, 16) === 'stack backtrace:') {
```
Comment (oli-obk) [inline lib/cargo.js:203]: is there a `starts_with` method for strings?

**Comment by oli-obk** [comment]
I wonder if all the file path checks should simply be replaced by checking whether the file exists...

lgtm w/ the nit

File: lib/cargo.js (hunk 1 above)
Comment (alygin) [inline lib/cargo.js:203]: Yeah, there's [String.startsWith()](http://www.w3schools.com/jsref/jsref_startswith.asp), but as far as I can guess, it hasn't been supported until recently. JavaScript and NodeJS are not my strong points, so I'm not sure if I can use it safely in Atom. If yes, I'll substitute this part of code.

Comment (oli-obk) [inline lib/cargo.js:203]: ah, fine with me then, leave it as is, we can always go around adjusting stuff later

**Comment by alygin** [comment]
If I understand correctly, you're talking about checks of paths that we meet in panic messages and their stack traces. If yes, then I see two main disadvantages of such approach:
1. The file existence checks are much more expensive than the syntax analysis. Its not a big deal for a single panic, but might become a problem if we have lots of test failures with deep stack traces that contain source code links.
2. Potentially, a user might have a `src` directory on the same level as his project. And that directory may contain the file with the same name as Rust points to. Then we'll end up with the link to that file instead of Rust's source code.

I tried lots of panic variations and project structures and it really looks safe to use `../src/` as the marker of the Rust's code and anything else as either absolute or relative path to our own source code. If `rustc` (or `cargo`) has to give us a link to a piece of our code that sits outside of the project directory, it always uses an absolute path and never a relative one that starts with `../`.

**Comment by oli-obk** [comment]
great analysis, thanks

**Pull request merged**